solutions/fully-configurable/variables.tf

```diff
@@ -69,132 +73,118 @@ variable "service_plan" {
+  validation {
+    condition     = var.kms_encryption_enabled ? var.existing_secrets_manager_kms_key_crn == null : true
+    error_message = "'kms_encryption_enabled' should be false if passing a value for 'existing_secrets_manager_kms_key_crn'."
+  }
 
-variable "skip_kms_iam_authorization_policy" {
-  type        = bool
-  description = "Set to true to skip the creation of an IAM authorization policy that permits all Secrets Manager instances in the resource group to read the encryption key from the KMS instance. If set to false, pass in a value for the KMS instance in the `existing_kms_instance_crn` variable. If a value is specified for `ibmcloud_kms_api_key`, the policy is created in the KMS account."
-  default     = false
-}
+  validation {
+    condition     = var.existing_kms_instance_crn != null ? var.kms_encryption_enabled : true
+    error_message = "If passing a value for 'existing_kms_instance_crn', you should set 'kms_encryption_enabled' to true."
+  }
 
-variable "existing_secrets_manager_kms_key_crn" {
-  type        = string
-  description = "The CRN of a Key Protect or Hyper Protect Crypto Services key to use for Secrets Manager. If not specified, a key ring and key are created."
-  default     = null
+  validation {
+    condition     = var.kms_encryption_enabled ? (var.existing_kms_instance_crn != null ? true : false) : true
+    error_message = "An 'existing_kms_instance_crn' is required if 'kms_encryption_enabled_bucket' is set to true."
+  }
 }
 
-########################################################################################################################
-# KMS properties required when creating an encryption key, rather than passing an existing key CRN.
-########################################################################################################################
-
 variable "existing_kms_instance_crn" {
   type        = string
   default     = null
   description = "The CRN of the KMS instance (Hyper Protect Crypto Services or Key Protect). Required only if `existing_secrets_manager_crn` or `existing_secrets_manager_kms_key_crn` is not specified. If the KMS instance is in different account you must also provide a value for `ibmcloud_kms_api_key`."
+
+  validation {
+    condition = anytrue([
+      can(regex("^crn:(.*:){3}(kms|hs-crypto):(.*:){2}[0-9a-fA-F]{8}(?:-[0-9a-fA-F]{4}){3}-[0-9a-fA-F]{12}::$", var.existing_kms_instance_crn)),
+      var.existing_kms_instance_crn == null,
+    ])
+    error_message = "The provided KMS instance CRN in the input 'existing_kms_instance_crn' in not valid."
+  }
+
+  validation {
+    condition     = var.existing_kms_instance_crn != null ? var.existing_secrets_manager_crn == null : true
+    error_message = "A value should not be passed for 'existing_kms_instance_crn' when passing an existing secrets manager instance using the 'existing_secrets_manager_crn' input."
+  }
+}
+
+variable "force_delete_kms_key" {
```

Comment:
Had a call with KMS team today - this needs to be hard coded to `true` in the DA main.tf. Beside it, can you leave a comment saying: "Force delete must be set to true, or the terraform destroy will fail since the service does not de-register itself from the key until the reclamation period has expired"